engine/src/main/java/org/pentaho/di/trans/steps/propertyoutput/PropertyOutput.java

```diff
@@ -170,40 +166,42 @@ public boolean processRow( StepMetaInterface smi, StepDataInterface sdi ) throws
         setOutputDone(); // signal end to receiver(s)
         return false;
       }
-
       if ( sendToErrorRow ) {
         putError( data.outputRowMeta, r, 1L, errorMessage, null, "PROPSOUTPUTO001" );
       }
     }
-
     return true;
   }
 
   public boolean checkSameFile() throws KettleException {
     return data.previousFileName.equals( data.filename );
   }
 
-  private void saveProcessingFile() {
-    data.previousFileName = data.filename;
-  }
-
   private void openNewFile() throws KettleException {
+    InputStream newInput = null;
     try {
       data.pro = new Properties();
       data.KeySet.clear();
 
       data.file = KettleVFS.getFileObject( data.filename, getTransMeta() );
       if ( meta.isAppend() && data.file.exists() ) {
-        data.pro.load( KettleVFS.getInputStream( data.file ) );
+        newInput = KettleVFS.getInputStream( data.file );
+        data.pro.load( newInput );
       }
-
       // Create parent folder if needed...
       createParentFolder();
-
-      saveProcessingFile();
-
+      //save processing file
+      data.previousFileName = data.filename;
     } catch ( Exception e ) {
       throw new KettleException( "Error opening file [" + data.filename + "]!", e );
+    } finally {
```

Comment:
let's wrap data.file file object retrieving with try-with-resources, and we won't be needing this finally section